src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -2103,70 +2028,72 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     nodes_to_collapse <- inputs.collapse_selected_nodes . map(move |_|
         (model_clone.selected_nodes(),empty_id)
     );
-    outputs.nodes_collapsed <+ nodes_to_collapse;
+    out.source.nodes_collapsed <+ nodes_to_collapse;
     }
 
 
     // === Set Node Expression ===
     frp::extend! { network
 
     set_node_expression_string  <- inputs.set_node_expression.map(|(id,expr)| (*id,expr.code.clone()));
-    outputs.node_expression_set <+ set_node_expression_string;
+    out.source.node_expression_set <+ set_node_expression_string;
 
 
 
     // ==================
     // === Move Nodes ===
     // ==================
 
-    mouse_pos_fix <- mouse.position.map(|p| Vector2(p.x,p.y));
-
+    mouse_pos <- mouse.position.map(|p| Vector2(p.x,p.y));
 
     // === Discovering drag targets ===
 
-    let main            = touch.nodes.down.clone_ref();
-    let main_pressed    = touch.nodes.is_down.clone_ref();
-    main_was_sel       <- main.map(f!((id) model.nodes.selected.contains(id)));
-    tgts_if_non_sel    <- main.map(|id|vec![*id]).gate_not(&main_was_sel);
-    tgts_if_sel        <- main.map(f_!(model.nodes.selected.items())).gate(&main_was_sel);
-    tgts               <- any(tgts_if_non_sel,tgts_if_sel);
-    eval tgts ((ids) model.disable_grid_snapping_for(ids));
-
-    main_pos_on_press       <- main.map(f!((id) model.node_position(id)));
-    mouse_pos_on_press      <- mouse_pos_fix.sample(&main);
-    mouse_pos_diff          <- mouse_pos_fix.map2(&mouse_pos_on_press,|t,s|t-s).gate(&main_pressed);
-    main_tgt_pos_rt_changed <- mouse_pos_diff.map2(&main_pos_on_press,|t,s|t+s);
-    just_pressed            <- bool (&main_tgt_pos_rt_changed,&main_pos_on_press);
-    main_tgt_pos_rt         <- any  (&main_tgt_pos_rt_changed,&main_pos_on_press);
+    let node_down      = touch.nodes.down.clone_ref();
+    let node_is_down   = touch.nodes.is_down.clone_ref();
+    node_in_edit_mode <- node_down.map2(&out.node_being_edited,|t,s| Some(*t) == *s);
+    node_was_selected <- node_down.map(f!((id) model.nodes.selected.contains(id)));
+    tgts_if_non_sel   <- node_down.map(|id|vec![*id]).gate_not(&node_was_selected);
+    tgts_if_sel       <- node_down.map(f_!(model.nodes.selected.items())).gate(&node_was_selected);
+    tgts_if_non_edit  <- any(tgts_if_non_sel,tgts_if_sel).gate_not(&node_in_edit_mode);
+    tgts_if_edit      <- node_down.map(|_|default()).gate(&node_in_edit_mode);
+    drag_tgts         <- any(tgts_if_non_edit,tgts_if_edit);
+    any_drag_tgt      <- drag_tgts.map(|t|!t.is_empty());
+    node_pos_on_down  <- node_down.map(f!((id) model.node_position(id)));
+    mouse_pos_on_down <- mouse_pos.sample(&node_down);
+    mouse_pos_diff    <- mouse_pos.map2(&mouse_pos_on_down,|t,s|t-s).gate(&node_is_down);
+    node_tgt_pos_rt   <- mouse_pos_diff.map2(&node_pos_on_down,|t,s|t+s);
+    just_pressed      <- bool (&node_tgt_pos_rt,&node_pos_on_down);
+    node_tgt_pos_rt   <- any  (&node_tgt_pos_rt,&node_pos_on_down);
```

Comment:
Can you please provide short description of all "selected, edited" neatpicks this code is implementing? It is quite hard to understand when not knowing its purpose.

Reply:
Ill add it tomorrow or in my next pr if you agree. lets talk